### PR TITLE
Update tooling

### PR DESCRIPTION
**Pull request by rowanmanning**
I would probably review this one commit at a time, and ignore the one that's just running ESLint's auto-fixer (as this changes all the whitespace and is probably a bit much to look at.

This updates the makefile, switches to ESLint, and updates the project to be more consistent with my other projects (as well as Pa11y/FT work).

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+module.exports = {
+	extends: '@rowanmanning/eslint-config/legacy',
+	globals: {
+		define: true
+	},
+	rules: {
+		'id-length': 'off',
```
Comment (JakeChampion) [inline .eslintrc.js]: Where are we using short identifiers?

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+		'no-void': 'off'
```
Comment (JakeChampion) [inline .eslintrc.js]: We can avoid the use of void by adding a second parameter to the IIFE wrapper and using that as the safe `undefined`.

File: Makefile.node [hunk 3]
```diff
@@ -64,25 +64,31 @@ verify-javascript: verify-eslint verify-jshint verify-jscs
 	@$(TASK_DONE)
 
 verify-dust:
-	@if [ -e .dustmiterc ]; then dustmite --path ./view && $(TASK_DONE); fi
+	@if [ -e .dustmiterc* ]; then dustmite --path ./view && $(TASK_DONE); fi
 
 verify-eslint:
-	@if [ -e .eslintrc ]; then eslint . && $(TASK_DONE); fi
+	@if [ -e .eslintrc* ]; then eslint . && $(TASK_DONE); fi
```
Comment (JakeChampion) [inline Makefile.node]: Ah yes, because they can be `.js` or `.json`, nice!

File: .eslintrc.js (hunk 2 above)
Comment (rowanmanning) [reply .eslintrc.js]: Is `undefined` ever safe? Because it's set outside of a strict mode function

File: .eslintrc.js (hunk 1 above)
Comment (rowanmanning) [reply .eslintrc.js]: Oh yeah I added that in temporarily. Will remove

File: Makefile.node [hunk 4]
```diff
@@ -96,15 +102,19 @@ test-unit:
...
+				istanbul cover $(NPM_BIN)/_mocha -- test/unit --recursive && $(TASK_DONE); \
+			else \
+				make test-unit; \
+			fi \
 		fi \
 	fi
 
 test-integration:
-	@if [ -d test/integration ]; then mocha test/integration --timeout $(INTEGRATION_TIMEOUT) --slow $(INTEGRATION_SLOW) && $(TASK_DONE); fi
+	@if [ -d test/integration ]; then mocha test/integration --timeout $(INTEGRATION_TIMEOUT) --slow $(INTEGRATION_SLOW) $(INTEGRATION_FLAGS) && $(TASK_DONE); fi
```
Comment (JakeChampion) [inline Makefile.node]: I don't think integration_flags is defined

Comment (rowanmanning) [reply Makefile.node]: Nope, it doesn't need to be. It's so that the Makefile that extends this can add in additional flags if it wants to

Comment (rowanmanning) [reply Makefile.node]: Proclaim doesn't actually have any integration tests

File: .eslintrc.js (hunk 2 above)
Comment (JakeChampion) [reply .eslintrc.js]: Should be safe, yes:
```js
(function(module, undef) {
	console.log(module, undef);
}({}))
```

Comment (rowanmanning) [reply .eslintrc.js]: Oooh! on it

**Pull request merged**